frontend/packages/console-dynamic-plugin-sdk/src/validation/ExtensionValidator.ts

```diff
@@ -29,13 +75,25 @@ export const collectCodeRefData = (extensions: Extension[]) =>
 export const findWebpackModules = (
   compilation: webpack.Compilation,
   exposedModules: ConsolePluginBuildMetadata['exposedModules'],
+  basePath: string,
 ) => {
   const webpackModules = Array.from(compilation.modules);
-
   return Object.keys(exposedModules).reduce((acc, moduleName) => {
+    const absolutePath =
+      basePath && guessModuleFilePath(path.resolve(basePath, exposedModules[moduleName]));
+
     acc[moduleName] = webpackModules.find((m) => {
-      const rawRequest = _.get(m, 'rawRequest') || _.get(m, 'rootModule.rawRequest');
-      return exposedModules[moduleName] === rawRequest;
+      /* first strategy: check if the module name matches the rawRequest */
+      // @ts-expect-error the real type is ConcatenatedModule || NormalModule, but it's not exported by webpack
```

Comment:
VS code shows `m` typed as `Module`, where does the `ConcatenatedModule || NormalModule` type union come from?

`compilation.modules` should be typed as `Set<webpack.Module>`.

That said, I now see the reason we used `_.get` to work around these TS errors :smiley:

Reply:
At runtime, `ConcatenatedModule` exists as a class but there's no type for it so that's why I found the `ConcatenatedModule || NormalModule` type union confusing.